modules/datahandling/projectrequests.go

```diff
@@ -0,0 +1,223 @@
+package datahandling
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+var projectRequestsSetup = false
+
+// initProjectRequests populates the requestMap from requestmap.go with the appropriate constructors for the project methods
+func initProjectRequests() {
+	if projectRequestsSetup {
+		return
+	}
+
+	// This isn't an ideal way of populating the map since there is so much duplicate code, but extracting this
+	// requires some kind of restructuring of the request hierarchy, which can come later because we can't come up
+	// with an obvious way to do it.
```

Comment:
If these methods just pass a *request  back, you might be able to create a helper method

Something like:

>func commonJson(req request, absReq abstractRequest) (request, error){
>	req.setAbstractRequest(absReq)
>	rawData := absReq.Data
>	err := json.Unmarshal(rawData, &req)
>	return req, err
>}

and 

>authenticatedRequestMap["ProjectCreateRequest"] = func(req abstractRequest) (request, error) {
>	return commonJson(new(projectCreateRequest), req)
>}